### ISSUE TITLE
Loop variables at end of loops are reported as unbound

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pyright#2420]
**Describe the bug**
If you have a `for` loop, the values of the loop variables are declared as unbound at the end of the loop

**To Reproduce**
See code below

**Expected behavior**
No error message.

**Screenshots or Code**
```python
adict = {"a": 1, "b": 2}

for i, k in enumerate(adict.keys()):
    print(i, k)

print("end of loop i is ", i)

for j in range(4):
    print(j)

print("end of loop ", j)

```
reports
```
"i" is unbound
"j" is unbound
```

**VS Code extension or command-line**
VS Code extension  pylance V2021.10.0

**Comment by erictraut** [comment]
I'll note that mypy doesn't run into this issue because it doesn't track unbound variables currently (although it's a highly-requested feature). If you find the unbound variable reporting to be too noisy in pyright, you can disable that diagnostic check (`reportUnboundVariable`).
